src/data/polynomial/degree/trailing_degree.lean

```diff
@@ -249,6 +249,23 @@ begin
   { exact le_nat_degree_of_ne_zero (mt trailing_coeff_eq_zero.mp hp) },
 end
 
+lemma nat_trailing_degree_mul_X_pow {p : polynomial R} (hp : p ≠ 0) (n : ℕ) :
+  (p * X ^ n).nat_trailing_degree = p.nat_trailing_degree + n :=
+begin
+  apply le_antisymm,
+  { apply nat_trailing_degree_le_of_ne_zero,
+    intro h,
+    apply mt trailing_coeff_eq_zero.mp hp,
```

Comment:
If you like you can combine these into a single `refine` statement, but it's certainly not necessary.

Reply:
I do like. Done.